### ISSUE TITLE
[LLVM6] Assertion triggered in `FastISel`

### ISSUE THREAD

**Issue by vchuravy** [JuliaLang/julia#26416]
Found during looking at #26398. At `O1` and `O0` we are using `FastISel` and we currently trigger a seqfault.
Since I won't have much time this week to look into this, I hope somebody else might have time.

I dumped the module in gdb and the specific function with `-print-before-all`, then used bugpoint to isolate the assertion warning.

```
; ModuleID = 'bugpoint-reduced-simplified.ll'
source_filename = "bugpoint-output-4b11196.bc"
target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128-ni:10:11:12"
target triple = "x86_64-unknown-linux-gnu"

; Function Attrs: sspstrong
define void @julia_bin_5346() #0 {
  br label %1

; <label>:1:                                      ; preds = %0
  br i1 undef, label %2, label %3

; <label>:2:                                      ; preds = %1
  ret void

; <label>:3:                                      ; preds = %1
  %4 = shl i8 undef, -1
  %5 = select i1 true, i8 0, i8 %4
  store i8 %5, i8* undef
  unreachable
}

attributes #0 = { sspstrong }

!llvm.module.flags = !{!0}

!0 = !{i32 1, !"Debug Info Version", i32 3}
```

**Comment by vtjnash** [comment]
What function did we generate this from?

**Comment by vchuravy** [comment]
According to the debug information https://github.com/JuliaLang/julia/blob/master/base/intfuncs.jl#L569-L579

**Comment by Keno** [comment]
Did you file this upstream?

**Comment by vchuravy** [comment]
No not yet.

**Comment by vchuravy** [comment]
Submitted upstream as https://bugs.llvm.org/show_bug.cgi?id=36731

**Comment by vchuravy** [comment]
fixed upstream and patch backported to #26398
